content/blog/django-model-fields-oracle-database-sequences/index.md

```diff
@@ -0,0 +1,221 @@
+---
+date: "2020-11-11T15:00:00Z"
+description: "Using Django model method decorators to set field values using Oracle Database sequences or other SQL with multi-database support."
+featuredImage: jon-tyson-UJN_XAg0ECI-unsplash.jpg
+featuredImageCredit: "Jon Tyson"
+featuredImageLink: "https://unsplash.com/photos/UJN_XAg0ECI"
+tags: ["django", "oracle", "oracle database"]
+title: Setting Django model fields automatically using Oracle Database sequences
+---
+
+_**Oracle Database not required!** The following will detail working with things like sequences in Oracle database but the decorator and general approach can be used in any database supported by Django._
+
+# TL;DR
+
+Use the `set_sql_for_field` decorator below on your model's `save()` method to get/set a sequence or any other arbitrary SQL.
+
+# An imperfect database for an imperfect world
+
+Sometimes when working with existing or legacy databases you might run into less-than-perfect implementations of things like primary keys.
+
+Could be that when the database was created the database management system (DBMS) didn't support things like `AUTO_INCREMENT` (MySQL) or `IDENTITY` (Oracle Database 12+). It could also be that those responsible had good reason to manually manage things or maybe didn't know of a different way of doing things.
+
+**Let's say for reasons beyond this blog post that you cannot alter the existing datbases/tables.**
```

Comment:
s/datbases/databases/

Reply:
Fixed, thanks for catching! Ran a full spell check in NeoVim, just to be sure I didn't miss anything else.